cms_monPerf

```diff
@@ -10,62 +10,58 @@
 # 4.      paging load, and
 # 5.      network utilization.
 
-max() {
-  printf "%d\n" "${@}" | sort -rn | head -1
-}
-
 NCPU=$(awk '/processor/ {nr++} END{print nr}' /proc/cpuinfo)
-LOADAVG=$(< /proc/loadavg)
-MEMINFO=$(< /proc/meminfo)
 
-LOAD_ARR=(${LOADAVG})
-LOAD5=$(echo "scale=0;${LOAD_ARR[1]}*100/${NCPU}" | bc -l)
+LOAD5=$(awk '{print int($2*100/'${NCPU}')}' /proc/loadavg)
 [[ "${LOAD5}" -gt "100" ]] && LOAD5="100"
 
 # procs -----------------------memory---------------------- ---swap-- -----io---- -system-- --------cpu--------
 #  r  b         swpd         free         buff        cache   si   so    bi    bo   in   cs  us  sy  id  wa  st
 CPU_UTIL_INFO=$(vmstat 1 2 | tail -n1)
 
 CPU=$(echo "${CPU_UTIL_INFO}" | awk '{print $(NF-3) + $(NF-4)  }') #'
-CPU_WA=$(echo "${CPU_UTIL_INFO}" | awk '{print $(NF-1)  }') #'
+#CPU_WA=$(echo "${CPU_UTIL_INFO}" | awk '{print $(NF-1)  }') #'
 
-MemTot=$(echo "${MEMINFO}" | awk '/MemTotal/ {print $(NF-1)}') #'
-MemFree=$(echo "${MEMINFO}" | awk '/MemFree/ {print $(NF-1)}') #'
-MemUsed=$((MemTot - MemFree))
-MEM=$(echo "scale=0;${MemUsed}*100/${MemTot}" | bc -l)
+MEM=$(awk '/MemTotal/{MEM_TOT=$(NF-1)}/MemFree/{MEM_FREE=$(NF-1)}END{print int((MEM_TOT-MEM_FREE)/MEM_TOT*100)}' /proc/meminfo)
 
 # should be replaced by IOwait/NCPU ?
 PGIO=0
 
-NET_UTIL_LIST=""
+NET_LOAD="0"
 IFACE_DIR="/sys/class/net"
-for iface in $(ls -1 --hide lo ${IFACE_DIR}) ; do
-  IS_UP=$(grep up ${IFACE_DIR}/${iface}/operstate)
-  [[ -z "${IS_UP}" ]] && continue
-  SPEED=$(< ${IFACE_DIR}/${iface}/speed)
-  SPEED_BYTES=$(echo "scale=0; ${SPEED} * 1000000/8 " | bc -l)
-
-  SET1=$(cat ${IFACE_DIR}/${iface}/statistics/{rx,tx}_bytes | tr '\n' ' ')
-  SET1_ARR=(${SET1})
-  RX1=SET1_ARR[0]
-  TX1=SET1_ARR[1]
+for iface in ${IFACE_DIR}/*; do
+  IFACE_NAME=${iface##*/}
+  [ "${IFACE_NAME}" = "lo" ] && continue
+  OPERSTATE=$(< ${iface}/operstate)
+  [ "$OPERSTATE" != "up" ] && continue
+  SPEED=$(< ${iface}/speed)
+  if [ ! $? -eq 0 ]; then
+    SPEED=1000
+  elif [ ! ${SPEED} -gt 0 ]; then
+    SPEED=1000
+  fi
```

Comment:
IMHO if the speed could not be read OR is invalid we should skip the interface  
also i just discovered that not physical devices are symlinks to paths with virtual ... so we should just skip them with this selection `[[ $(realpath ${iface}) =~ virtual ]] && continue`

Reply:
For a `virtio-net` interface, you get:
```
eth0 -> ../../devices/pci0000:00/0000:00:03.0/virtio0/net/eth0
```
but the speed can not be read (invalid argument). So to support network interfaces in VMs, we need to assume something, or VMs will be without network. 
But I fully agree we should skip virtual interfaces :wink:.